### PR TITLE
ENT-2879 | Added the support in CourseEntitlementFulfillmentModule cl…

### PR DESCRIPTION
**Pull request by hasnain-naveed**
…ass to create enterprise related records.

### CODE REVIEW

**Review by moconnell1453 — APPROVED** [review]
Great consolidation of code into the new mixin!

File: ecommerce/enterprise/tests/test_mixins.py [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+
+    def test_get_enterprise_customer_cost_for_line(self):
+        """
+        Test correct values for discount percentage are evaluated and rounded.
+        """
+        line_price = Decimal('199.00')
+        effective_discount_percentage = Decimal('0.001027742658353086344768502165')
+
+        # pylint: disable=protected-access
+        actual = EnterpriseDiscountMixin()._get_enterprise_customer_cost_for_line(
```
Comment (hasnain-naveed) [inline ecommerce/enterprise/tests/test_mixins.py]: `_get_enterprise_customer_cost_for_line` is a static method. We can call it with class.

File: ecommerce/enterprise/tests/test_mixins.py [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+        """
+        Test correct values for discount percentage are evaluated when discount type is FIXED.
+        """
+        enterprise_contract_metadata = EnterpriseContractMetadata(
+            discount_type=EnterpriseContractMetadata.FIXED,
+            discount_value=Decimal('12.3456'),
+            amount_paid=Decimal('12000.00')
+        )
+        # pylint: disable=protected-access
+        actual = EnterpriseDiscountMixin()._calculate_effective_discount_percentage(enterprise_contract_metadata)
```
Comment (hasnain-naveed) [inline ecommerce/enterprise/tests/test_mixins.py]: same.

File: ecommerce/enterprise/tests/test_mixins.py [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+        """
+        Test correct values for discount percentage are evaluated when discount type is PERCENTAGE.
+        """
+        enterprise_contract_metadata = EnterpriseContractMetadata(
+            discount_type=EnterpriseContractMetadata.PERCENTAGE,
+            discount_value=Decimal('12.3456'),
+            amount_paid=Decimal('12000.00')
+        )
+        # pylint: disable=protected-access
+        actual = EnterpriseDiscountMixin()._calculate_effective_discount_percentage(enterprise_contract_metadata)
```
Comment (hasnain-naveed) [inline ecommerce/enterprise/tests/test_mixins.py]: same.

File: ecommerce/enterprise/tests/test_mixins.py (hunk 1 above)
Comment (saleem-latif) [reply ecommerce/enterprise/tests/test_mixins.py]: nit: You can call static methods directly on classes like
```
EnterpriseDiscountMixin._get_enterprise_customer_cost_for_line(
```

**Review by saleem-latif — APPROVED** [review]
Looks Good 👍

File: ecommerce/enterprise/tests/test_mixins.py (hunk 1 above)
Comment (hasnain-naveed) [reply ecommerce/enterprise/tests/test_mixins.py]: Already done.

**Review by HammadAhmadWaqas — APPROVED** [review]

**Pull request merged**